waterbox/gpgx/cinterface/cinterface.c

```diff
@@ -657,10 +661,70 @@ GPGX_EX int gpgx_init(const char* feromextension,
 	config.mg = settings->MidGain; //100;
 	config.hg = settings->HighGain; //100;
 
-	config.ym2612         = settings->YM2612SoundChip == YM2612_Vanilla;
-	config.ym2413         = settings->YM2413SoundChip == YM2413_Mame;
-	config.ym3438         = settings->YM2612SoundChip == YM2612_Nuked;
-	config.opll           = settings->YM2413SoundChip == YM2413_Nuked;
+
+// enum SMSFMSoundChipType
+// {
+// 	YM2413_MAME = 0,
+// 	YM2413_NUKED = 1
+// };
+
+// enum GenesisFMSoundChipType
+// {
+// 	YM2612_MAME_DISCRETE = 0,
+// 	YM2612_MAME_ENHANCED = 1,
+// 	YM2612_NUKED = 2,
+// 	YM3438_MAME = 4,
+// 	YM3438_NUKED = 5
+// };
+
+    // Selecting FM Sound chip to use for SMS / GG emulation
+    switch (settings->SMSFMSoundChip)
+	{
+	  case YM2413_DISABLED: 
+	    config.opll = 0;
+		config.ym2413 = 0;
+        break;
+
+      case YM2413_MAME: 
+	    config.opll = 0;
+		config.ym2413 = 1;
+        break;
+
+	  case YM2413_NUKED: 
+	    config.opll = 1;
+		config.ym2413 = 0;
+        break;
+	}
+
+	// Selecting FM Sound chip to use for Genesis / Megadrive / CD emulation
+    switch (settings->SMSFMSoundChip)

```

Comment:
Copypaste error?

Reply:
I have an issue with tabbing/spaces with my IDE I still haven't been able to figure out. Feel free to fix